### PR TITLE
support rtl

### PR DESCRIPTION
**Pull request by xingstarx**
I think this feature is need for this repo

### CODE REVIEW

**Comment by xingstarx** [comment]
Do you think it is necessary to use VIewCompat.getPaddingStart()? 
After my test I can think of two ways is ok

File: pinentryedittext/src/main/java/com/alimuzaffar/lib/pin/PinEntryEditText.java [hunk 1]
```diff
@@ -220,21 +220,21 @@ protected void onSizeChanged(int w, int h, int oldw, int oldh) {
...
         mCharBottom = new float[(int) mNumChars];
-        int startX = getPaddingLeft();
+        int startX = ViewCompat.getPaddingStart(this);
         int bottom = getHeight() - getPaddingBottom();
         int rtlFlag = 1;
         final boolean isLayoutRtl = TextUtilsCompat.getLayoutDirectionFromLocale(Locale.getDefault()) == ViewCompat.LAYOUT_DIRECTION_RTL;
         if (isLayoutRtl) {
             rtlFlag = -1 * rtlFlag;
-            startX = (int) (getWidth() - getPaddingRight() - mCharSize);
+            startX = (int) (getWidth() - ViewCompat.getPaddingEnd(this) - mCharSize);
```
Comment (alphamu) [inline pinentryedittext/src/main/java/com/alimuzaffar/lib/pin/PinEntryEditText.java]: This is now incorrect, paddingStart for RTL is the right side so it should still be width - paddingStart.
Ideally, we should support both, if paddingStart is not set but paddingLeft is, we should use it. For now, just make it work with start and end or revert out to working with left and right and I can add the missing functionality.

**Comment by xingstarx** [comment]
You are right , I will  modify it in this weekend. 
thanks

**Comment by xingstarx** [comment]
afater my test ,when set paddingLeft or paddingStart in layout  xml  ,VIewCompat.getPaddingStart() value is same , when paddingLeft and paddingStart all set, the value we get by VIewCompat.getPaddingStart() is paddingStart

and below Build.VERSION_CODES.JELLY_BEAN_MR1 , not support rtl

and now , you need code review my code ...

If there is a bug, please tell me

File: pinentryedittext/src/main/java/com/alimuzaffar/lib/pin/PinEntryEditText.java [hunk 2]
```diff
@@ -216,16 +220,29 @@ protected void onSizeChanged(int w, int h, int oldw, int oldh) {
...
         }
         mLineCoords = new RectF[(int) mNumChars];
         mCharBottom = new float[(int) mNumChars];
-        int startX = getPaddingLeft();
+        int startX;
         int bottom = getHeight() - getPaddingBottom();
+        int rtlFlag = 1;
+        final boolean isLayoutRtl = TextUtilsCompat.getLayoutDirectionFromLocale(Locale.getDefault()) == ViewCompat.LAYOUT_DIRECTION_RTL;
+        // can't support rtl
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.JELLY_BEAN_MR1) {
```
Comment (alphamu) [inline pinentryedittext/src/main/java/com/alimuzaffar/lib/pin/PinEntryEditText.java]: Look at the code for `ViewCompat`, it returns `view.getPaddingLeft()` for everything below JellybeanMR1 so the condition to check for older phones is not needed. `ViewCompat` has a base implementation for `getPaddingStart()` that return `getPaddingLeft()`. This is then overwritten for the JRMR1 implementations where it returns view.getPaddingStart(). So you were correct the first time with this. I recommend removing this if condition and just using the conditions inside the else block.

    if (isLayoutRtl) {
        rtlFlag = -1;
        startX = (int) (getWidth() - ViewCompat.getPaddingStart(this) - mCharSize);
    } else {
        rtlFlag = 1;
        startX = ViewCompat.getPaddingStart(this);
    }

**Comment by xingstarx** [comment]
should I need to rebase commit record?

**Comment by alphamu** [comment]
Looks good, I'll accept the changes.
This only solves the problem of Alignment, however, the text is still display Left to Right.
We could easily change this to display text Right to Left, however, some languages like Arabic, are written Right to Left, but their numbers are written Left to Right. Then other languages like Hebrew have text and numbers written right to left.

Since the widget supports both text and numbers, I'm not sure how to handle this currently.

**Comment by alphamu** [comment]
These changes have been deployed in release v1.1 along with updating the code to build for Android API 24.

**Comment by xingstarx** [comment]
In our team, we don't deal with special text and numbers
Our products are Shou (https://play.google.com/store/apps/details?id=tv.shou.android)

it support rtl

As you said, it only solves the problem of Alignment.

I have an idea, FaceBook AccountKit with PinEntryEditText has a similar function
https://developers.facebook.com/docs/accountkit/android
You can run Facebook AccountKit sample , check Is it necessary to support text and numbers.
and I will check next Monday

ps:  my english is very poor,hope you do not mind.

**Pull request merged**